core/src/fields.rs

```diff
@@ -576,20 +577,40 @@ impl MongoFields {
                             let schema_collection =
                                 db.collection::<Document>(SQL_SCHEMAS_COLLECTION);
 
-                            // If the schema for `collection_name` isn't found, default to an empty schema.
-                            let schema_doc: Document = schema_collection
-                                .find_one(doc! {
+                            let match_agg_pipeline = vec![doc! {
+                                "$match": {
                                     "_id": &collection_name
-                                })
+                                }
+                            }];
+
+                            let mut schema_coll_agg_response: Vec<Document> = schema_collection
+                                .aggregate(match_agg_pipeline)
                                 .await
                                 .map_err(Error::QueryExecutionFailed)?
-                                .unwrap_or({
-                                    log::warn!("No schema was found for collection `{}`. It will be assigned\
-                                    an empty schema. Hint: Generate schemas for your collections.", collection_name);
+                                .try_collect::<Vec<Document>>()
+                                .await
+                                .map_err(Error::QueryExecutionFailed)?;
+
+                            if schema_coll_agg_response.len() > 1 {
```

Comment:
```suggestion
                            if schema_coll_agg_response.len() > =1 {
```
There could be only one, no?